### PR TITLE
Support Hot/Warm Archicture Better

### PR DESCRIPTION
**Pull request by DandyDeveloper**
My original implementation didn't fully consider this scenario. 

By default, for hot/warm you add an aliased release. The problem is this has no discovery service / uses a different template name. This override allows the user for hot/warm to define the cluster discovery service for extra deployments.

### CODE REVIEW

**Comment by kalvinnchau** [comment]
Hey @DandyDeveloper! Thanks for the PR! 
It's just @DerekHeldtWerle and I that are maintaining this repo at the moment, we're gonna test it out in our cluster and then we'll get this merged.

**Comment by DerekHeldtWerle** [comment]
Hey @DandyDeveloper this is great! Would you mind adding a section to the README about how to deploy the hot/warm architecture (e.g. a basic example _warm/code_ `values.yml` file) and detail which values are required and why that's the case? This'll help alleviate the number of future inquires into why a users installation may or may not be supported in their setup/environment.

**Comment by DandyDeveloper** [comment]
@DerekHeldtWerle Done. Can you take a look?

File: open-distro-elasticsearch-kubernetes/helm/README.md [hunk 1]
```diff
@@ -353,6 +353,61 @@ elasticsearch:
...
+
+After the `requirements.yaml` is configured. Another instance of the chart can be defined in the `values.yaml` to provide the details of the warm environment. The benefit to using the alias attribute, is you can define as many different other environments as you want with some additional configuration options. 
+
+There are some key attributes here that should be considered.
+ - `elasticsearch.discoveryOverride`  Override for the service name deployed by the original chart alias. (Default: `*namespace*-*alias*-discovery`)
+ - `kibana.enabled: false`  Disables and prevent another instance of Kibana
+ - `elasticsearch.master.enabled: false`  Disables and prevent another instance of ES Master
+ - `elasticsearch.client.enabled: false`  Disables and prevent another instance of ES Ingest/Client
+
+*Note that the config is a snippet showing how you would enable index routing. Please find the [full configuration example here](#elasticsearch.yml-Config)*
```
Comment (DerekHeldtWerle) [inline open-distro-elasticsearch-kubernetes/helm/README.md]: `(#elasticsearch.yml-Config)` -> `(#elasticsearchyml-config)`

File: open-distro-elasticsearch-kubernetes/helm/README.md [hunk 2]
```diff
@@ -353,6 +353,61 @@ elasticsearch:
...
+    version: ^0.0.1
+    repository: file://../opendistro-es/
+  # Warm Nodes
+  - name: opendistro-es
+    version: ^0.0.1
+    repository: file://../opendistro-es/
+    alias: opendistro-es-data-warm
+```
+
+After the `requirements.yaml` is configured. Another instance of the chart can be defined in the `values.yaml` to provide the details of the warm environment. The benefit to using the alias attribute, is you can define as many different other environments as you want with some additional configuration options.
```
Comment (DerekHeldtWerle) [inline open-distro-elasticsearch-kubernetes/helm/README.md]: `configured. Another` -> `configured, another`

**Review by DerekHeldtWerle — CHANGES_REQUESTED** [review]
Just some very minor things, but otherwise looks great!

**Comment by DandyDeveloper** [comment]
@DerekHeldtWerle Done!

**Pull request merged**